sdv/constraints/tabular.py

```diff
@@ -107,6 +119,13 @@ def _validate_inputs(cls, **kwargs):
                 ]
                 raise AggregateConstraintsError(errors)
 
+        def __reduce__(self):
+            return (
+                _RecreateCustomConstraint(),
+                (is_valid_fn, transform_fn, reverse_transform_fn),
+                self.__dict__
+            )
```

Comment:
Interesting... I did something similar the first time I went over this issue with pickling the constraints, here is the `pr`: https://github.com/sdv-dev/SDV/pull/944/ and funny enough this was still using `pickle` as seen [here](https://github.com/sdv-dev/SDV/blob/6411ba1baabd3a45eef4425433a0c2b1bb01d4d9/sdv/tabular/base.py#L864). However, after that we had an issue sampling from it after loading which leads to this `pr`: https://github.com/sdv-dev/SDV/issues/984 .

I think that at some point we used `dill` but `dill` was just able to save the model but not sample from it. Therefore we went for something more 'powerful' such as `cloudpickle` which at that time solved the problem with saving a model and loading - sampling from it.

PS: I think that there may be no need of having the `_RecreateCustomConstraint` 😅